applications/unit_tests/detail/adapter.cpp

```diff
@@ -12,26 +12,22 @@ TEST_CASE("Adapter")
 {
     llri::Instance* instance = detail::defaultInstance();
 
-    std::vector<llri::Adapter*> adapters;
-    REQUIRE_EQ(instance->enumerateAdapters(&adapters), llri::result::Success);
-
-    for (llri::Adapter* adapter : adapters)
-    {
+    detail::iterateAdapters(instance, [=](llri::Adapter* adapter) {
```

Comment:
any particular reasin you auto capture by value? and not by reference?

Reply:
well i don't know if any actual captures are being done, but yes generally copying pointers is faster than copying the full data XD

Reply:
yeah if it's only the instance then you could either pass it to the lambda, since the calling function has the instance anyways. or explicitly capture it by value. since the instance is a pointer it doesn't make much sense to capture that by reference